src/startupscriptgenerator/python/scriptgenerator.go

```diff
@@ -91,68 +91,85 @@ func logReadDirError(logger *common.Logger, path string, err error) {
 }
 
 // Builds the commands to setup the Python packages, using virtual env or a package folder.
-func (gen *PythonStartupScriptGenerator) getPackageSetupCommand() string {
+func (gen *PythonStartupScriptGenerator) getPackageSetupCommand(buildManifest common.BuildManifest) string {
 	scriptBuilder := strings.Builder{}
-	manifesFilePath := filepath.Join(gen.SourcePath, consts.BuildManifestFileName)
-	// If a manifest file is present, it takes precedence.
-	if common.FileExists(manifesFilePath) {
-		virtualEnvVarPath := filepath.Join(gen.SourcePath, "$virtualEnvName")
-		scriptBuilder.WriteString("echo \"Using '" + consts.BuildManifestFileName + "'.\"\n")
-		scriptBuilder.WriteString(". ./" + consts.BuildManifestFileName + "\n")
-
-		if !gen.SkipVirtualEnvExtraction {
+	
+	// Values in manifest file takes precedence over values supplied at command line
+	virtualEnvironmentName := buildManifest.VirtualEnvName
+	if virtualEnvironmentName == "" {
+		virtualEnvironmentName = gen.VirtualEnvironmentName
+	}
+	packageDirName := buildManifest.PackageDir
+	if packageDirName == "" {
+		packageDirName = gen.PackageDirectory
+	}
+
+	if virtualEnvironmentName != "" {
+		scriptBuilder.WriteString("virtualEnvName=\"" + virtualEnvironmentName + "\"\n")
+		virtualEnvDir := filepath.Join(gen.SourcePath, virtualEnvironmentName)
+
+		// If virtual environment was not compressed or if it is compressed but mounted using a zip driver,
+		// we do not want to extract the compressed file
+		if buildManifest.CompressedVirtualEnvFile == "" || gen.SkipVirtualEnvExtraction {
+			if common.PathExists(virtualEnvDir) {
+				// We add the virtual env site-packages to PYTHONPATH instead of activating it to be backwards compatible with existing
+				// app service implementation. If we activate the virtual env directly things don't work since it has hardcoded references to
+				// python libraries including the absolute path. Since Python is installed in different paths in build and runtime images,
+				// the libraries are not found.
+				scriptBuilder.WriteString("virtualEnvDir=\"" + virtualEnvDir + "\"\n")
+				virtualEnvCommand := getVirtualEnvironmentCommand()
+				scriptBuilder.WriteString(virtualEnvCommand)
+
+			} else {
+				packageDirName = "__oryx_packages__"
```

Comment:
@gpcastro I will add a comment here but basically this is the place where we set the packagedir to the old default package directory and line 156 gets evaluated for this case.

Reply:
This value repeats in both a C# project and this Go project - maybe consolidate them in the YAML? (not necessarily in this PR)

Reply:
yes, i have a bug filed earlier for this which will handle this scenario